packages/opentelemetry-metrics/src/export/aggregators/histogram.ts

```diff
@@ -0,0 +1,96 @@
+/*!
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Aggregator, Point } from '../types';
+import { HrTime } from '@opentelemetry/api';
+import { hrTime } from '@opentelemetry/core';
+
+export type Checkpoint = {
+  buckets: {
+    boundaries: number[];
+    counts: number[];
+  };
+  sum: number;
+  count: number;
+};
+
+/** Basic aggregator which calculates a Sum from individual measurements. */
+export class HistogramAggregator implements Aggregator {
+  private _lastCheckpoint: Checkpoint;
+  private _currentCheckpoint: Checkpoint;
+  private _lastCheckpointTime: HrTime = [0, 0];
+  private _boundaries: number[];
+
+  constructor(boundaries: number[]) {
+    if (boundaries === undefined || boundaries.length === 0) {
+      throw new Error(`HistogramAggregator should be created with boundaries.`);
+    }
+    this._boundaries = boundaries.sort();
+    this._lastCheckpoint = this._newEmptyCheckpoint();
+    this._currentCheckpoint = this._newEmptyCheckpoint();
+  }
+
+  get sum() {
+    return this._lastCheckpoint.sum;
+  }
+
+  get count() {
+    return this._lastCheckpoint.count;
+  }
+
+  get checkpoint() {
```

Comment:
AFAIK only `toPoint` method is used by exporter to export the data, why do we need to expose `sum`, `count` and `checkpoint`? I think we should expose as minimal as possible to avoid breaking changes later.

Reply:
As same as above, i've tried to match the go implementation

Reply:
Edit: From the spec [here](https://github.com/jmacd/opentelemetry-specification/blob/jmacd/draft_metric_sdk_spec/specification/sdk-metric.md#L25), we should expose a `toCheckpoint` rather than `toPoint`, so thats why i've added this one. However i think we need to rename it in other aggregators, so i'll use `toPoint` too and we'll make a PR to rename them later.

Reply:
I think if it is called `toPoint` in other aggregators, we should use the same here. If it needs to be renamed, it should be renamed everywhere (including here) in a separate PR. Don't want to be in a state where we have both correct and incorrect naming in the same package.